src/jvmMain/kotlin/io/github/oshai/kotlinlogging/logback/internal/LogbackLoggerFactory.kt

```diff
@@ -0,0 +1,27 @@
+package io.github.oshai.kotlinlogging.logback.internal
+
+import ch.qos.logback.classic.Logger
+import ch.qos.logback.classic.LoggerContext
+import ch.qos.logback.classic.spi.LogbackServiceProvider
+import io.github.oshai.kotlinlogging.KLogger
+
+internal object LogbackLoggerFactory {
+
+  private val logbackServiceProvider = createLogbackServiceProvider()
+
+  private fun createLogbackServiceProvider(): LogbackServiceProvider {
+    val logbackServiceProvider = LogbackServiceProvider()
+    logbackServiceProvider.initialize()
+    return logbackServiceProvider
+  }
+
+  /** get a java logger by name. Logback relies on SLF4J logger factory */
+  internal fun jLogger(name: String): Logger =
+    logbackServiceProvider.loggerFactory.getLogger(name) as Logger
+
+  /** wrap java logger based on location awareness */
```

Comment:
actually here there is no difference in location awerness so this can be removed

Reply:
Removed. And renamed methods and fields a bit.